torchmultimodal/modules/layers/quantisation.py

```diff
@@ -66,33 +67,34 @@ def quantise(self, x_flat: Tensor):
         distances = torch.cdist(x_flat, self.embedding.weight, p=2.0) ** 2
 
         # Encoding - select closest embedding vectors
-        encoding_indices = torch.argmin(distances, dim=1)
+        codebook_indices = torch.argmin(distances, dim=1)
 
         # Quantise
-        quantised_flat = self.embedding(encoding_indices)
+        quantised_flat = self.embedding(codebook_indices)
 
         # Straight through estimator
         quantised_flat = x_flat + (quantised_flat - x_flat).detach()
 
-        return quantised_flat
+        return quantised_flat, codebook_indices
 
     def forward(self, x: Tensor):
         # Reshape and flatten encoder output for quantisation
         x_flat, permuted_shape = self._preprocess(x)
 
         # Quantisation via nearest neighbor lookup
-        quantised_flat = self.quantise(x_flat)
+        quantised_flat, codebook_indices = self.quantise(x_flat)
+        self.quantised_vectors = quantised_flat
+        self.codebook_indices = codebook_indices
```

Comment:
why do we set instance variables inside forward? i thought we can do away with get_quantised based on discussion with Lan or did I understand wrong?

Reply:
My understanding was that it may be necessary to retrieve the quantised vectors without having to do a forward pass again, @langong347

Reply:
actually I understood this incorrectly, you are right the forward should just return the quantised vectors + codebook indices. will update this according to the design doc.